src/JoinEqWithDTMsg.chpl

```diff
@@ -376,6 +377,43 @@ module JoinEqWithDTMsg
         return new MsgTuple(repMsg, MsgType.NORMAL);
     }// end joinEqWithDTMsg()
 
+    proc codeMappingMsg(cmd: string, msgArgs: borrowed MessageArgs, st: borrowed SymTab): MsgTuple throws {
+        jeLogger.debug(getModuleName(),getRoutineName(),getLineNumber(),"Mapping codes from categories");
+
+        var catName = msgArgs.getValueOf("categories");
+        var queryName = msgArgs.getValueOf("query");
+
+        var catSym = getSegString(catName, st);
+        var querySym = getSegString(queryName, st);
+
+        var querySize = querySym.size;
+        var catSize = catSym.size;
+        var codes: [0..#querySize] int = -1;
+
+        var (catOff, catVal) = catSym[0..#catSize];
+        var (queryOff, queryVal) = querySym[0..#querySize];
+        
+        // For all indexes in the query segString, find the matching index in the categories segString
+        // Exponential Growth - break once a match is found
+        coforall i in 0..#querySize {
+            for x in 0..#catSize {
+                if queryVal[i] == catVal[x]{
+                    codes[i] = x;
```

Comment:
might need @Ethan-DeBandi99's input on the `coforall` since he's got more experience. My knee jerk reaction is this should be a `forall` but i don't know for sure

I'm not sure if it would be more efficient but one idea of a more vectorized way to do the inner loop is something like the below. This is just pseudocode i haven't tried running it
```chapel
forall (val, code) in zip(queryVal, codes) {
  // create bool array of locations where we match the queryval
  var equalsVal = val == catVal;
  // find the first place we match
  var (maxVal, maxLoc) = maxloc reduce zip(equalsVal, equalsVal.domain);
  if maxVal {
    code = maxLoc;
  }
}
```

Reply:
the `forall` is actually the outerloop (replacing the `coforall`). The inner loop/ for loop is being replaced by the `==` and `maxloc reduce`

Reply:
The coforall here won't actually do much because you are iterating over a local object which won't allow for any parallelism. So this is effectively fully serial, breaking into a single `forall` will at least allow for task parallel operations.

Reply:
That was my fault for misreading it. 
`10**6` with a `forall` outer-loop
```
-------------------------------------- benchmark 'Inner Join': 1 tests --------------------------------------
Name (time in s)         Min      Max     Mean  StdDev   Median     IQR  Outliers     OPS  Rounds  Iterations
-------------------------------------------------------------------------------------------------------------
bench_inner_join     16.8480  17.4797  17.1619  0.2655  17.1079  0.4548       2;0  0.0583       5           1
-------------------------------------------------------------------------------------------------------------
```